packages/template-retail-react-app/app/components/island/README.md

```diff
@@ -0,0 +1,50 @@
+
+Ultimately, the goal is to reduce the load on the browser’s main thread, which can otherwise easily be overwhelmed by the sheer volume of JavaScript to be processed on initial page load. Achieving an efficient compute reduction requires decisions during the development process on how to best break down a page into smaller, meaningful islands of interactivity.
+
+## React’s Selective Hydration ≠ Partial Hydration
+
+React 18, with [Suspense](https://react.dev/reference/react/Suspense), laid the foundation to officially support concepts such as [selective hydration](https://github.com/reactwg/react-18/discussions/37). However, while the Islands Architecture concept treats partial and selective hydration as interchangeable terms, React’s understanding of selective hydration differs in key aspects.
+
+Suspense _delays the **rendering** of certain components altogether_ and instead enables the display of specified fallback content while the actual content is loading.
+
+When using Suspense, certain contents are therefore not rendered on the server at all, but instead asynchronously on the client at a later time only — with all the associated implications for accessibility and SEO. Depending on the runtime used, the boundaries can be fluid as to when the asynchronous loading of the actual/final content begins — whether already on the server or only on the client — but the outcome remains the same: **Suspense enables asynchronous rendering, not asynchronous hydration.**
+
+In addition to Suspense, [Server Components](https://react.dev/reference/rsc/server-components) are now another official concept in React for addressing the all-or-nothing problem with hydration. However, as the name suggests, Server Components are components that are rendered exclusively on the server and therefore don’t receive any downstream client-side hydration.
+
+## `<Island/>` Component
+
+Unlike Suspense, the `<Island/>` component explicitly **encourages the synchronous rendering of HTML on the server** as much as possible. And instead of delaying the rendering of certain contents, it allows the **explicit delay of the hydration process itself**.
+
+This delay can be achieved using various `hydrateOn` strategies:
+
+1. `hydrateOn={'load'}`: Useful for high-priority UI elements that need to be interactive as quickly as possible. Default behavior if `hydrateOn` is not defined.
+2. `hydrateOn={'idle'}`: Useful for lower-priority UI elements that don’t need to be immediately interactive. Hydration occurs during the [browser’s next available idle period](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback).
+3. `hydrateOn={'visible'}`: Useful for lower-priority UI elements that don’t need to be immediately interactive. Hydration occurs once the component has [entered the user’s viewport](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver). As a result, such an element doesn’t get hydrated at all if the user never sees it.
```

Comment:
@seckardt thanks for the reference. I hadn't heard of this API, I learned a bunch today 📚 

Very powerful for hidden elements like the header navigation which definitely introduce lots of hydration bloat while also being initially invisible